SHS-6106: Dashboard: Siteimprove integration

laura-johnson opened the pull request:
# Summary
This adds a module hs_siteimprove that is a wrapper for the SiteImprove API. The module currently contains a block to view broken links that I have added to the dashboard.

Work for two tickets is in this PR:
[Dashboard: SiteImprove Integration](https://app.clickup.com/t/36718269/SHS-6106)
[Dashboard: Broken Links in Content Pages](https://app.clickup.com/t/36718269/SHS-5977)

We'll need to add environment variables to the acquia environments for this to work on there.

If testing locally you can add the following to your settings.php file:

 // Set the SiteImprove API key and username.
```
$config['hs_siteimprove.settings'] = [
  'api_key' => getenv('SITEIMPROVE_API_KEY') ?: '[key]',
  'username' => getenv('SITEIMPROVE_USERNAME') ?: '[username]',
];
```

You can find the value for the username in this ticket: https://app.clickup.com/t/36718269/SHS-6106
You can find the value for the api in this ticket: https://app.clickup.com/t/36718269/SHS-6002

# Testing steps
- [ ] Go to the dashboard at /admin/dashboard and verify you see the block, called SiteImprove Broken Links Report
- [ ] I also added a controller for testing where you should see the site details from SiteImprove at /admin/config/services/siteimprove/details

- For it to get the id of the site you're using, the first argument in the URL must match the site https://africanstudies.suhumsci.loc/

It is working on the tugboat preview for west. To preview, use the cli:

https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com/

View the SiteImprove Broken Links Report here: https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com/admin/dashboard

Note that as per Albert's feedback, the report should display 20 links sorted by number of broken links

The link at the bottom of the blocks points to https://my2.siteimprove.com/QualityAssurance/[site_id]/Links/Pages/1/UnsuccessfulClicks/Desc?pageSize=100


## Need Review By (Date)
_[When does this need to be reviewed by? '10/30', 'asap', etc.]_

## Urgency
_['low', 'medium', 'high', etc.]_

## Steps to Test
1. _[First testing step]_
2. ...

# Review Tasks

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Are PHP functions and variables in `snake_case` and not `camelCase`?
- [ ] Does Drupal code follow [Drupal Coding Standards](https://www.drupal.org/docs/develop/standards/php/php-coding-standards)?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided?

### Code security
- [ ] Is all [user input properly sanitized when rendered](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

### General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

dalin- committed:
feat(SHS-5913): Initial install of Dashboard module

ahughes3 committed:
Merge branch 'develop' into SHS-5913_Dashboard-initiative

cienvaras committed:
Merge branch '11.6.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

mariannuar committed:
SH-5972: Use default Drupal layouts in Dashboard (#1697)

* feat(shs-5972): use default drupal layouts in dashboard

* feat(shs-5972): use DashboardSectionStorage instead

* feat(shs-5972): use DashboardSectionStorage instead

* fix(shs-5972): small fixes in code

* fix(shs-5972): small fixes in code

* Update docroot/modules/humsci/hs_layouts/hs_layouts.info.yml

Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

---------

Co-authored-by: Mari Nez <mariannuar@gmail.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
Merge branch '11.6.4-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

cienvaras committed:
feat(shs-6052): export base dashboard

cienvaras committed:
Merge branch '11.7.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

mariannuar committed:
SHS-6035: Dashboard: Initial design implementation (#1719)

* feat(shs-6035): add dashboard initial design implementation

* feat(shs-6035): add dashboard initial design implementation

* feat(shs-6035): add dashboard initial design implementation

* feat(shs-6035): finish adding general dashboard design implementation

* feat(shs-6035): finish adding general dashboard design implementation

* feat(shs-6035): add styles to hover states in row background, edit actions, and links

* feat(shs-6035): move dashboard styles to a new file and attach file only to the dashboard

* fix(shs-6035): fix preg_match

* fix(shs-6035): add empty line at the end of file

* fix(shs-6035): change order of .yml file

* feat(shs-6035): add style table thead tr for all view-content in dashboards

* fix(shs-6035): remove version in yml file and use route name instead of regexes

* SHS-5981: Site Dashboard: Active Site Editors (#1721)

* feat(shs-5981): add dashboard site active editors block

* feat(shs-5981): add styles to site active editors block

* feat(shs-5981): add empty line at the end of the css file

* feat(shs-5981): add a more link in the footer of the active site editors block

---------

Co-authored-by: Mari Nez <mariannuar@gmail.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

* chore(shs-6035): add active site editors block to test dashboard

---------

Co-authored-by: Mari Nez <mariannuar@gmail.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
Merge branch '11.7.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

cienvaras committed:
fix(shs-5913): coding standards error

callinmullaney committed:
SHS-5978: Site Dashboard: Missing Alt Text (#1722)

* feat: add dashboard block for missing alt text media and attempt to expose decorative value in views

* feat(shs-5978): update missing alt text block view

* fix(shs-5978): remove unused files

* fix(shs-5978): update view title

* fix(shs-5978): add headers to the missing alt text block

* chore(shs-5978): add missing alt text block to dashboard

* chore(shs-5978): restore missing files

---------

Co-authored-by: Callin Mullaney <57088-callinmullaney@users.noreply.drupalcode.org>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

codechefmarc committed:
SHS-5979 and SHS-5982: Add dashboard views and custom view code (#1730)

* feat(SHS-5979 and SHS-5982): Add dashboard views and custom view code

* chore(SHS-5979 and SHS-5982): Temp add chemistry as a Tugboat site

* chore(SHS-5979 and SHS-5982): Temp add chemistry CI settings

* feat(SHS-5979): Add pager for duplicate people - possible fix for timeout on Tugboat

* feat(SHS-5979 and SHS-5982): Change no results text

* chore(shs-5979): add latest edits and duplicate people blocks to test dashboard

---------

Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
chore(shs-5913): remove chemistry tugboat config

mariannuar committed:
SHS- 6004: Dashboard: Pages with the Most issues from editoria11y (#1734)

* feat(shs-6004): update pages with the most issues block from editoria11y

* chore(shs-6004): add editoria11y block to dashboard

* fix(shs-6004): remove duplicated requirement in exported dashboard config

---------

Co-authored-by: Mari Nez <mariannuar@gmail.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
fix(shs-5913): update dashboard exported config

codechefmarc committed:
SHS-6055: Dashboard: Importer block: People (#1747)

* feat(SHS-6055): Add importers block with people importer info table

* fix(SHS-6055): Fix typo in dependency

* fix(SHS-6055): Change to snake case, fix empty orphan action setting

* feat(SHS-6055): Add no results message when no people importers are found

* refactor(SHS-6055): Change name of some variables

* chore(shs-6055): add importers block to test dashboard

* feat(SHS-6055): Add changes from Albert review

---------

Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
Merge branch '11.8.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

cienvaras committed:
Merge branch '11.8.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

codechefmarc committed:
SHS-5980: Dashboard: Notifications/Announcements to Site Editors (#1745)

* feat(SHS-5979 and SHS-5982): Add dashboard views and custom view code

* chore(SHS-5979 and SHS-5982): Temp add chemistry as a Tugboat site

* chore(SHS-5979 and SHS-5982): Temp add chemistry CI settings

* feat(SHS-5979): Add pager for duplicate people - possible fix for timeout on Tugboat

* feat(SHS-5979 and SHS-5982): Change no results text

* chore(shs-5979): add latest edits and duplicate people blocks to test dashboard

* feat(SHS-5980): Add announcements block

* fix(SHS-5980): Fix linting error

* fix(shs-5980): remove tugboat chemistry config

* fix(SHS-5980): Fix caching and empty CSV rows

* fix(SHS-5980): Minor fixes from code review

* feat(SHS-5980): Remove title column

---------

Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
Merge branch '11.8.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

mariannuar committed:
SHS-5975: Dashboard: University Policies block (#1768)

* feat(shs-5975): add university policies block to dashboard

* fix(shs-5975): fix php error

* fix(shs-5975): fix php error

---------

Co-authored-by: Mari Nez <mariannuar@gmail.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
Merge branch '11.8.2-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

cienvaras committed:
Merge branch '11.8.2-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

codechefmarc committed:
SHS-6056: Dashboard: Importer block: Events (#1755)

* feat(SHS-6055): Add importers block with people importer info table

* fix(SHS-6055): Fix typo in dependency

* fix(SHS-6055): Change to snake case, fix empty orphan action setting

* feat(SHS-6055): Add no results message when no people importers are found

* refactor(SHS-6055): Change name of some variables

* chore(shs-6055): add importers block to test dashboard

* chore(SHS-6056): WIP - research only code at the moment to figure out how to get Localist data into the dashboard

* feat(SHS-6055): Add changes from Albert review

* feat(SHS-6056): WIP - getting event importer info into table

* feat(SHS-6056): WIP - finished all filters, places, audiences, depts/groups. Working on bookmark

* chore(SHS-6056): Remove kint for now

* feat(SHS-6056): Finalize event importer table, clean up code, refactor, and add documentation

* fix(SHS-6056): Fix linting errors, refactor code for better readability

* chore(SHS-6056): codeclimate told me to use spaces for strict_types, phpcs says not to.

* chore(SHS-6056): Temp switch to Four Kitchens fork of stanford_fields repo for testing

* chore(SHS-6056): Attempt to get Four Kitchens repo to pull on CI

* refactor(SHS-6056): Change importers to use the plugin system for ease of adding additional importers and to seperate code

* fix(SHS-6056): Fix linting errors

* fix(SHS-6056): Fix linting errors

* fix(shs-6056): change title and remove column from dashboard event importer info block

---------

Co-authored-by: Marc Berger <marc@fourkitchens.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

nJim committed:
feat(shs-6057): add course importer to dashboard (#1776)

* feat(SHS-6055): Add importers block with people importer info table

* fix(SHS-6055): Fix typo in dependency

* fix(SHS-6055): Change to snake case, fix empty orphan action setting

* feat(SHS-6055): Add no results message when no people importers are found

* refactor(SHS-6055): Change name of some variables

* chore(shs-6055): add importers block to test dashboard

* chore(SHS-6056): WIP - research only code at the moment to figure out how to get Localist data into the dashboard

* feat(SHS-6055): Add changes from Albert review

* feat(SHS-6056): WIP - getting event importer info into table

* feat(SHS-6056): WIP - finished all filters, places, audiences, depts/groups. Working on bookmark

* chore(SHS-6056): Remove kint for now

* feat(SHS-6056): Finalize event importer table, clean up code, refactor, and add documentation

* fix(SHS-6056): Fix linting errors, refactor code for better readability

* chore(SHS-6056): codeclimate told me to use spaces for strict_types, phpcs says not to.

* chore(SHS-6056): Temp switch to Four Kitchens fork of stanford_fields repo for testing

* chore(SHS-6056): Attempt to get Four Kitchens repo to pull on CI

* refactor(SHS-6056): Change importers to use the plugin system for ease of adding additional importers and to seperate code

* fix(SHS-6056): Fix linting errors

* fix(SHS-6056): Fix linting errors

* feat(shs-6057): add course importer to dashboard

* feat(shs-6057): linting fix

* fix(shs-6056): change title and remove column from dashboard event importer info block

* fix(shs-6057): update dashboard courses importer block

---------

Co-authored-by: Marc Berger <marc@fourkitchens.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

cienvaras committed:
Merge branch '11.9.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

laura-johnson committed:
shs-6106 clean up branch

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

laura-johnson committed:
fix: shs-6106 remove testing identifier

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

laura-johnson committed:
fix: shs-6106 remove check for acquia env when setting keys

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

laura-johnson committed:
Update global.settings.php

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

laura-johnson committed:
fix: shs-6106 update global.settings.php get env variables

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

laura-johnson committed:
feat: shs-6106 add config settings to west settings.php

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

laura-johnson committed:
feat: shs-6106 update config settings

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

laura-johnson committed:
feat: shs-6106 update config settings

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-colorful-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://hs-traditional-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://history-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com
- https://west-cgbxqmnu99lecjrrmzbkvv1egejcvznw.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f

Visual Diffs:
- https://dashboard.tugboatqa.com/67e44ca8908d23476b0a4e2f/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

laura-johnson committed:
Update settings.php

laura-johnson committed:
shs-6106 update global settings siteimprove config

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

laura-johnson committed:
fix: shs-6106 update siteimprove config keys

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

laura-johnson committed:
fix: shs-6106 add conditional to set siteimprove values

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

laura-johnson committed:
fix: shs-6106 remove test url

cienvaras commented on docroot/modules/humsci/hs_siteimprove/README.md:
```diff
@@ -0,0 +1,30 @@
+## INTRODUCTION
```
Can you please replace this placeholder content with a real Readme. Doesn't need to be anything complex, just some basic information on what the module does.

cienvaras commented on docroot/modules/humsci/hs_siteimprove/hs_siteimprove.routing.yml:
```diff
@@ -0,0 +1,17 @@
+    _form: '\Drupal\hs_siteimprove\Form\SiteImproveSettingsForm'
+    _title: 'SiteImprove API Settings'
+  requirements:
+    _permission: 'access administration pages'
```
I think we need a custom permission for this. There are roles that have basic access to the admin pages that probably shouldn't be able to change these settings.

cienvaras commented on docroot/modules/humsci/hs_siteimprove/src/Plugin/Block/BrokenLinksBlock.php:
```diff
@@ -0,0 +1,182 @@
+        '#title' => $this->t('View Full Broken Links Report'),
+        '#url' => $this->getBrokenLinksReportUrl(),
+        '#attributes' => [
+          'class' => ['siteimprove-full-report-link'],
```
We're standardizing the appearance of the "View more" links, let's change the class to re-use existing styles:
```suggestion
          'class' => ['button', 'button--primary'],
```

cienvaras commented on docroot/modules/humsci/hs_siteimprove/src/Plugin/Block/BrokenLinksBlock.php:
```diff
@@ -0,0 +1,182 @@
+          'class' => ['siteimprove-full-report-link'],
+          'target' => '_blank',
+        ],
+        '#prefix' => '<div class="siteimprove-full-report">',
```
Also here:

```suggestion
        '#prefix' => '<div class="more-link">',
```

cienvaras reviewed (changes requested):
@laura-johnson Looks pretty good, thanks! There are a couple of minor things to fix. Let me know if you have any questions.

laura-johnson commented:
Great feedback, thanks @cienvaras I'll make these changes

laura-johnson committed:
feat: shs-6106 add caching, permission, readme, adjust classes

laura-johnson committed:
fix: shs-6106 remove extra css

laura-johnson committed:
feat: shs-6106 add permanent caching for getSites()

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

laura-johnson commented:
@cienvaras I addressed the issues you raised:
- Added info to the README
- Added caching to the broken links block that expires after 5 minutes
- I also added caching to the getSites() call that is permanent (until cache clear), because we don't anticipate the sites changing frequently, this will decrease the call time after the block cache clears
- Added an 'administer siteimprove' permission for the routes
- Updated the CSS classes used in the block to the standardized ones

cienvaras commented on docroot/modules/humsci/hs_siteimprove/src/SiteImprove.php:
```diff
@@ -0,0 +1,308 @@
+      }
+
+      // Cache for 5 minutes.
+      $this->cache->set($cid, $pages, time() + 300);
```
@laura-johnson You're caching the block already. Why cache the data too?

laura-johnson replied:
@cienvaras You're right, I was experimenting with caching there too but we don't need it as it's just for the one block which is cached already. I removed it.

cienvaras reviewed (commented):
@laura-johnson Looks great, thanks for the fixes. Just left a question in the code.

laura-johnson committed:
fix: shs-6106 remove unneeded caching from getPagesWithBrokenLinks() call

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

mariannuar committed:
SHS-6135: Dashboard: Importers block design updates (#1797)

* feat(shs-6135): dashboard importers block design updates

* feat(shs-6135): dashboard importers block design updates

* feat(shs-6135): dashboard importers block design updates

* fix(shs-6135): PHPCS errors

---------

Co-authored-by: Mari Nez <mariannuar@gmail.com>

cienvaras committed:
Merge branch '11.9.2-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

cienvaras reviewed (approved):
@laura-johnson LGTM! Thanks for all the fixes.

@ahughes3 Ready for you

nJim committed:
SHS-6058: Dashboard: Other importers (#1796)

* feat(shs-6058): add other importers to dashboard

* feat(shs-6058): add other importers to dashboard

ahughes3 commented:
Confirmed that we want to reduce the broken links list down to 5

ahughes3 commented on docroot/modules/humsci/hs_siteimprove/src/Plugin/Block/BrokenLinksBlock.php:
```diff
@@ -0,0 +1,190 @@
+        $actions,
+      ];
+
+      // Limit the number of rows to 20.
```
Can we reduce the list to 5 items

dalin- committed:
SHS-6125: Dashboard: Importers Updates (#1803)

* feat(shs-6058): add other importers to dashboard

* feat(shs-6058): add other importers to dashboard

* feat(SHS-6125): show last imported time for each dashboard importer block

* feat(SHS-6125): always show 'last imported' as the last column

---------

Co-authored-by: nJim <jamesvomero@gmail.com>
Co-authored-by: Andrés Díaz Soto <andres.diaz.soto@gmail.com>

cienvaras committed:
SHS-6131 - Dashboard: Pre-release fixes and improvements (#1792)

* feat(shs-6131): update dashboard module to 2.0.0

* chore(shs-6131): add dashboard module as a dependency for hs_dashboard

* chore(shs-6131): update dashboard layout

* chore(shs-6131): change view title 'Duplicate People' -> 'Duplicate Profiles'

* chore(shs-6131): change column title in 'My Latest Edits' view

* chore(shs-6131): update date format in dashboard blocks

* fix(shs-6131): update dashboard styles / fix 'View more' link styles

* chore(shs-6131): assign view dashboard permission to contributor and site manager roles

* feat(shs-6131): add back to top element to dashboard

* fix(shs-6131): import dashboard config before assigning permissions

* fix(shs-6131): store announcements csv url in config instead of hardcoded variable

* fix(shs-6131): change how announcements block is rendered / add caching

* fix(shs-6131): fix how dasbboard permissions are defined

* fix(shs-6131): minor fixes

laura-johnson committed:
feat: shs-6106 update broken links call and limit to 5 items

nJim committed:
merge develop and resolve conflicts

cienvaras committed:
Merge branch '11.10.1-release' of github.com:SU-HSDO/suhumsci into SHS-5913_Dashboard-initiative

cienvaras committed:
Merge branch 'SHS-5913_Dashboard-initiative' of github.com:SU-HSDO/suhumsci into shs-6106-siteimprove-integration

laura-johnson committed:
fix: shs-6106 limit block rows to 5

cienvaras committed:
chore(shs-5977): add siteimprove broken links block to dashboard

cienvaras committed:
Merge branch 'shs-6106-siteimprove-integration' of github.com:SU-HSDO/suhumsci into shs-6106-siteimprove-integration

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

cienvaras commented:
@ahughes3 This one is ready for you to review again. We never discussed the position of the broken links, so for now I added it below the "Pages with the most accessibility issues". Let me know if we should update the position.

ahughes3 commented:
For consistency the "view more" link should be - `View full broken links report`

cienvaras committed:
Merge branch '11.11.1-release' of github.com:SU-HSDO/suhumsci into shs-6106-siteimprove-integration

cienvaras committed:
fix(shs-6106): update exported dashboard configuration

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

cienvaras committed:
fix(shs-6106): capitalization of siteimprove broken links block view more link

cienvaras committed:
fix(shs-6106): increase page size in siteimprove site request

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1790-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-colorful-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://hs-traditional-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://history-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com
- https://west-9iufhlvoldghmjoeertxykqupeyb4kpb.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67ed9830e026d48ed338de53

cienvaras commented:
@ahughes3 I updated and fixed conflicts in this PR, it should be ready for a final test. I had to increase the number of sites being pulled in the initial request to SiteImprove from 200 to 500, because the account being used to access the API (`ahughes3`) now has more sites associated and the previous limit wasn't preventing from pulling all sites. This shouldn't have a big impact on performance, because that initial request is being cached, but still it would be a good idea to use a SiteImprove account that only has access to the platform sites when deploying to production.

ahughes3 commented:
@cienvaras Is it possible to sort or filter the list of sites that are returned to look for HSDP or [HSDP] in the title?

All of the HSDP sites will have that and would cut the list down to under 200.

![image](https://github.com/user-attachments/assets/ab920d8e-3c20-4212-8c8a-725992b4e599)

cienvaras commented:
@ahughes3 No, it's not possible to filter by text in that API endpoint. However, it's possible to pass a "Group ID" argument, maybe it's possible to configure this in SiteImprove, so all the platform site are in a single group?

joegl commented on docroot/modules/humsci/hs_siteimprove/hs_siteimprove.module:
```diff
@@ -0,0 +1,6 @@
+
+/**
+ * @file
+ * Primary module hooks for stanford_siteimprove module.
```
Is this file still necessary? And should it say `hs_siteimprove` instead of `stanford_siteimprove`?

joegl commented:
@ahughes3 @laura-johnson This is merged and available on staging for testing. Let me know if you'd like me to put the environment variables for the API key and username in the secrets file on all 3 environments. I saw the API key in ClickUp ticket but did not find the username and will need that (if you put it in ClickUp just let me know where to find it!). Thanks
